lib/elixir/lib/enum.ex

```diff
@@ -717,30 +717,61 @@ defmodule Enum do
   """
   @spec fetch(t, integer) :: {:ok, element} | :error
   def fetch(enumerable, index)
+
+  def fetch(enumerable, index) when is_integer(index) and index < 0 do
+    case Enumerable.count(enumerable) do
+      {:ok, count} when (count + index) < 0 ->
+        :error
+
+      {:ok, count} ->
+        fetch(enumerable, count + index)
+
+      {:error, _} ->
+        enumerable
+        |> reverse([])
+        |> fetch_list(abs(index + 1))
```

Comment:
I thought of this as well, but i thought it may not be accepted.
actually it should be `-1`.
code updated